snuba/pipeline/query_pipeline.py

```diff
@@ -76,3 +81,51 @@ def build_planner(
         self, query: Query, settings: QuerySettings
     ) -> QueryPlanner[ClickhouseQueryPlan]:
         raise NotImplementedError
+
+
+class QueryPipelineStage(Generic[Tin, Tout]):
```

Comment:
does it matter to make this class abstract?

Reply:
We should as it helps enforce a common interface. Added!